### ISSUE TITLE
NoReturn function causes finally block to be marked as unreachable

### ISSUE THREAD

**Issue by francovs** [microsoft/pylance-release#592]
## Environment data

-   Language Server version: 2020.11.0 (pyright 9d8e431b)
-   OS and version: Windows 10 Home
-   Python version: Python 3.9.0 64-bit

## Expected behaviour

A finally block should be considered reachable even if it can only be reached through exception cleanup.

## Actual behavior

Pylance considers the finally block unreachable if the end of the associated try block is unreachable; code appears grayed out in the editor.

## Code Snippet / Additional information

Bug can be reproduced with the following code:

```python
def infinite_loop_that_raises():
    x = 0
    while True:
        x += 1
        if x > 1000 : raise ValueError

try:
    infinite_loop_that_raises()
    print ("Unreachable code")  # Pylance correctly marks as unreachable
finally:
    print ("Cleanup. Definitely reachable.")  # Pylance *incorrectly* marks as unreachable
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.11.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020111-11-november-2020
